resource/js/tab-hierarchy.js

```diff
@@ -0,0 +1,223 @@
+              if (concept.narrower) {
+                // corresponding concept node in hierarchy tree
+                const conceptNode = parent.children.find(c => c.uri === concept.uri)
+                // children of current concept
+                const children = concept.narrower.map(c => {
+                  return { uri: c.uri, label: c.label, hasChildren: c.hasChildren, children: [], isOpen: false }
+                })
+                // set children of current concept as children of concept node
+                conceptNode.children = children
+                conceptNode.isOpen = children.length !== 0
+                // push concept node to parent queue
+                parents.push(conceptNode)
+              }
+            }
+          }
+
+          this.loading = false
+          console.log(this.hierarchy)
+        })
+    },
+    loadChildren (concept) {
+      // load children only if concept has children but they have not been loaded yet
+      if (concept.children.length === 0 && concept.hasChildren) {
+        fetch('rest/v1/' + SKOSMOS.vocab + '/children?uri=' + concept.uri + '&lang=' + SKOSMOS.lang)
+          .then(data => {
+            return data.json()
+          })
+          .then(data => {
+            console.log(data)
+            for (const c of data.narrower) {
+              concept.children.push({ uri: c.uri, label: c.prefLabel, hasChildren: c.hasChildren, children: [], isOpen: false })
+            }
+            console.log(this.hierarchy)
+          })
+      }
+    }
+  },
+  template: `
+    <div v-click-tab-hierarchy="handleClickHierarchyEvent">
+      <ul v-if="!loading">
+        <template v-for="c in hierarchy">
+          <tab-hier :concept="c" @load-children="loadChildren($event)"></tab-hier>
+        </template>
+      </ul>
+      <template v-else>Loading...</template><!-- Add a spinner or equivalent -->
+    </div>
+  `
+})
+
+tabHierApp.directive('click-tab-hierarchy', {
+  beforeMount: (el, binding) => {
+    el.clickTabEvent = event => {
+      binding.value() // calling the method given as the attribute value (handleClickHierarchyEvent)
+    }
+    document.querySelector('#hierarchy').addEventListener('click', el.clickTabEvent) // registering an event listener on clicks on the hierarchy nav-item element
+  },
+  unmounted: el => {
+    document.querySelector('#hierarchy').removeEventListener('click', el.clickTabEvent)
+  }
+})
+
+tabHierApp.component('tab-hier', {
+  props: ['concept'],
+  emits: ['loadChildren'],
+  inject: ['partialPageLoad'],
+  methods: {
+    handleClickOpenEvent (concept) {
+      concept.isOpen = !concept.isOpen
+      this.$emit('loadChildren', concept)
+    },
+    handleClickConceptEvent (event, concept) {
+      concept.isOpen = true
+      this.$emit('loadChildren', concept)
+      this.partialPageLoad(event, this.getHref(concept.uri))
+    },
+    loadChildrenRecursive (concept) {
+      this.$emit('loadChildren', concept)
+    },
+    getHref (uri) {
```

Comment:
I think this method will eventually be needed in other tabs as well, so it could be extracted into a general utility method. But for now it can live here until we need it elsewhere.